Utilities/Log.cs

```diff
@@ -1,206 +1,213 @@
-        /// </summary>
-        /// <param name="Message"> Message to write. </param>
-        private static void Write(string Message)
-        {
-            if (Log.Destination == WriteDestination.ALL || Log.Destination == WriteDestination.CONSOLE) 
-            {
-                lock (ConsoleLock) { Console.Write(Message); }
-            }
-            if (Log.Destination == WriteDestination.ALL || Log.Destination == WriteDestination.FILE)
-            {
-                lock (FileLock) { LogFile.Write(Message); }
-            }
-        }
-
-        /// <summary> Writes a string and applies a new line in the console with a carriage return and line feed. </summary>
-        /// <param name="Message"> Message to write. </param>
-        private static void WriteLine(string Message) { Write(Message + "\r\n"); }
-
-        /// <summary> Write destination. </summary>
-        public enum WriteDestination { CONSOLE, FILE, ALL }
-
-        /// <summary> The subsystem where the error occured, for use in output filtering. </summary>
-        public enum Source { ALL, MOTORS, NETWORK, GUI, SENSORS, CAMERAS, SUBSYSTEM, HARDWAREIO, OTHER }
-
-        /// <summary> How important the log entry is. </summary>
-        /// Debug = Used for program flow debugging and light troubleshooting (e.g. "Starting distance sensor handler")
-        /// Information = Used for troubleshooting (e.g. "Distance sensor detected successfully")
-        /// Warning = When an issue arises, but functionality sees little to no impact (e.g. "Distance sensor took longer than expected to find value")
-        /// Error = When an issue arises that causes significant loss of functionality to one system (e.g. "Distance sensor unreachable")
-        /// Fatal = When an issue arises that causes loss of functionality to multiple systems or complete shutdown (e.g. Current limit reached, shutting down all motors)
-        public enum Severity { DEBUG, INFO, WARNING, ERROR, FATAL }
-
-    }
-}
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.IO;
+
+namespace Scarlet.Utilities
+{
+    public static class Log
+    {
+        // Override these in your implementation.
+        // OutputType determines which system you see output from.
+        // OutputLevel determines the minimum severity required for a message to be output.
+
+        private static WriteDestination P_Destination = WriteDestination.CONSOLE;
+
+        /// <summary> Sets where log output is directed. </summary>
+        public static WriteDestination Destination 
+        {
+            get { return P_Destination; }
+            set 
+            { 
+                P_Destination = value;
+                if (value == WriteDestination.ALL || value == WriteDestination.FILE) { CreateLogFile(); }
+            }
+        }
+
+        private static String FileDestination = "ScarletLog-" + DateTime.Now.ToString("yy-MM-dd-hh-mm-ss-tt");
+
+        public static String LogFileDestination

```

Comment:
I see no reason for this to be a variable/Property combination, as you are not defining custom behaviour.